lumicks/pylake/kymo.py

```diff
@@ -586,18 +589,40 @@ def estimate_bead_edges(
             any prior deprecation notice! If you use this functionality keep a close eye on the
             changelog for any changes that may affect your analysis.
 
+        There are two algorithms to determine bead edges:
+
+            - "brightness": brightness based bead edge detection
+            - "template": template correlation-based bead edge detection
+
+        1. Brightness
+
+        Searches for bead edges by summing along the temporal direction and removing small features.
+        The result is smoothed and peaks are detected. Bead edges are found by checking where the
+        fluorescence drops below a threshold.
+
+        2. Template
+
+        Downsamples the kymograph to a specific number of frames (specified with the optional
+        parameter `num_time_frames`). For each frame, we traverse the scan line pixel by pixel,
+        extracting a template at each pixel. This template is then correlated with the next frame.
+        This process is repeated until all frames have been processed and a 2D matrix is acquired.
+        This matrix is then summed over the temporal axis obtaining a similarity score over time.
+        This score can be interpreted as a measure for how long a template taken at that particular
+        location is present. The outer maxima of this curve provide the bead centers.
+
+        Once the centers are found, we start searching from a bead_diameter from the center until
+        the fluorescence drops below a baseline level.
+
         Parameters
         ----------
         bead_diameter : float
             Rough estimate for the bead size (microns).
+        algorithm : str
+            Which algorithm to use. Valid options are "brightness", "template".
         channel : 'red', 'green', 'blue', optional
             Channel to use for bead detection.
         plot : bool, optional
             Plot result
-        threshold_percentile : int
-            Percentile to drop down to before accepting that we have left the bead area. Higher
-            values will make the bounds go closer to the bead edge but risk the algorithm failing
-            or having high background.
         **kwargs
             Forwarded to bead finding function.
```

Comment:
I wonder if we should even expose this in the public function at first? It would require users digging into the docstrings for the internal functions anyway. Only downside is it'd make things more difficult for us for testing, but we can just use the private functions directly. 

If we really want to expose it, I think I'd prefer to expose all of the individual parameters with defaults rather than `kwargs` and only the relevant ones get forwarded. What do you think?

Reply:
I've added them to the docstring. The reason I held off on it was because I didn't want to have to do breaking changes quickly. But I think the alpha warning covers that this functionality may change quickly.

The defaults should be good for most cases.